starport/pkg/jsonfile/jsonfile.go

```diff
@@ -0,0 +1,303 @@
+package jsonfile
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/pkg/errors"
+
+	"github.com/tendermint/starport/starport/pkg/tarball"
+)
+
+const (
+	keySeparator = "."
+)
+
+type (
+	// JSONFile represents the file
+	JSONFile struct {
+		file        readWriteSeeker
+		tarballPath string
+		url         string
+		updates     map[string][]byte
+	}
+	// UpdateFileOption configures file update.
+	UpdateFileOption func(map[string][]byte)
+
+	writeTruncate interface {
+		Truncate(size int64) error
+	}
+	readWriteSeeker interface {
+		io.ReadWriteSeeker
+		Close() error
+		Sync() error
+	}
+)
+
+var (
+	// ErrParamNotFound parameter not found into json
+	ErrParamNotFound = errors.New("parameter not found")
+	// ErrInvalidValueType invalid value type
+	ErrInvalidValueType = errors.New("invalid value type")
+)
+
+// New creates a new JSONFile
+func New(file readWriteSeeker) *JSONFile {
+	return &JSONFile{
+		updates: make(map[string][]byte),
+		file:    file,
+	}
+}
+
+// FromPath parse JSONFile object from path
+func FromPath(path string) (*JSONFile, error) {
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0600)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot open the file")
+	}
+	return New(file), nil
+}
+
+// FromURL fetches the file from the given URL and returns its content.
+func FromURL(ctx context.Context, url, path, tarballFileName string) (*JSONFile, error) {
+	// TODO create a cache system to avoid download genesis with the same hash again
+
+	// Download the file from URL
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// Remove the old file if exists and create a new one
+	if err := os.RemoveAll(path); err != nil {
+		return nil, err
+	}
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0600)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create the genesis file")
```

Comment:
I think there should not be any "genesis" reference because this is a logically separated package.